play-dl/index.ts

```diff
@@ -24,22 +24,17 @@ export async function stream_from_info(
     else return await yt_stream_info(info, cookie);
 }
 
-export async function validate(url: string): Promise<string | boolean> {
-    if (url.indexOf('spotify') !== -1) {
-        const check = sp_validate(url);
-        if (check) {
-            return 'sp_' + check;
-        } else return check;
-    } else if (url.indexOf('soundcloud') !== -1) {
-        const check = await so_validate(url);
-        if (check) {
-            return 'so_' + check;
-        } else return check;
+export async function validate(url: string): Promise<"so_playlist" | "so_track" | "sp_track" | "sp_album" | "sp_playlist" | "yt_video" | "yt_playlist" | false> {
+    let check;
+    if (url.includes('spotify')) {
```

Comment:
Please change this back to indexOf. Since includes checks first whether check is a regexp or string. So it takes little bit more time in calculation.

Reply:
Okay then.